### PR TITLE
The observer app is up and running localling with newly created docke…

### PR DESCRIPTION
**Pull request by ali-tm-amin**

### CODE REVIEW

**Review by scottaubrey — CHANGES_REQUESTED** [review]
On the whole it seems to work (start up), but I've not yet been able to get it running with data.

There are a few quality of live improvements for the dev experience here that would be worthwhile doing

File: .gitignore [hunk 1]
```diff
@@ -15,3 +15,4 @@ app.cfg
 uwsgi.ini
 ingestion-lax.log.json
 /src/core/prod_settings.py
+.env
```
[the remark this answers is not in the document]
Comment (ali-tm-amin) [reply .gitignore:18]: I will create document file.

File: Dockerfile [hunk 2]
```diff
@@ -0,0 +1,34 @@
...
+RUN pip install --no-cache-dir -r requirements.txt
+
+# Copy the observer application code
+COPY . /app
+RUN chmod +x /app/install.sh
+
+# Expose port (if the application runs a web server)
+EXPOSE 8000
+
+# COPY src/manage.py  /app/manage.py
```
[the remark this answers is not in the document]
Comment (ali-tm-amin) [reply Dockerfile]: Removed!

File: docker-compose.yaml [hunk 3]
```diff
@@ -0,0 +1,42 @@
...
+    - SERVICES=sqs
+    - AWS_DEFAULT_REGION=eu-east-1
+
+  app:
+    depends_on:
+      postgres:
+        condition: service_healthy
+      localstack:
+        condition: service_healthy
+      # migrate:
```
[the remark this answers is not in the document]
Comment (ali-tm-amin) [reply docker-compose.yaml]: Removed!

File: .gitignore (hunk 1 above)
[the remark this answers is not in the document]
Comment (scottaubrey) [reply .gitignore:18]: Thanks, that would be helpful.

**Comment by scottaubrey** [comment]
> On the whole it seems to work (start up), but I've not yet been able to get it running with data.
> 
> There are a few quality of live improvements for the dev experience here that would be worthwhile doing

@ali-tm-amin Just as a follow up - Using a simple config, I was able to import the eLife corpus into the config with SQLite DB using this command: `docker compose exec app ./manage.sh load_from_api`

After completing, it returns stuff! 

```
> curl http://localhost:8000/report/latest-articles
<?xml version='1.0' encoding='UTF-8'?>
<rss xmlns:dc="http://purl.org/dc/elements/1.1/" xmlns:webfeeds="http://webfeeds.org/rss/1.0" xmlns:atom="http://www.w3.org/2005/Atom" xmlns:content="http://purl.org/rss/1.0/modules/content/" version="2.0">
  <channel>
    <title>eLife: latest articles</title>
    <link>https://elifesciences.org</link>
    <description>All of the latest articles published at eLife, including in-progress POA (publish-on-accept) articles.</description>
    <atom:link href="https://observer.elifesciences.org/report/latest-articles" rel="self"/>
    <docs>http://www.rssboard.org/rss-specification</docs>
    <generator>observer (using python-feedgen)</generator>
    <language>en</language>
    <lastBuildDate>Tue, 21 May 2024 19:36:30 +0000</lastBuildDate>
    <webfeeds:analytics id="G-xxxxxxxxxx" engine="GoogleAnalytics"/>
    <item>
      <title>Inhibition of miR-199b-5p reduces pathological alterations in osteoarthritis by potentially targeting &lt;i&gt;Fzd6&lt;/i&gt; and &lt;i&gt;Gcnt2&lt;/i&gt;</title>
      <link>https://elifesciences.org/articles/92645</link>
      <description>Osteoarthritis (OA) is a degenerative disease with a high prevalence in the elderly population, but our understanding of its mechanisms remains incomplete. Analysis of serum exosomal small RNA sequencing data from clinical patients and gene expression data from OA patient serum and cartilage obtained from the GEO database revealed a common dysregulated miRNA, miR-199b-5p. In vitro cell experiments demonstrated that miR-199b-5p inhibits chondrocyte vitality and promotes extracellular matrix degradation. Conversely, inhibition of miR-199b-5p under inflammatory conditions exhibited protective effects against damage. Local viral injection of miR-199b-5p into mice induced a decrease in pain threshold and OA-like changes. In an OA model, inhibition of miR-199b-5p alleviated the pathological progression of OA. Furthermore, bioinformatics analysis and experimental validation identified &lt;i&gt;Gcnt2&lt;/i&gt; and &lt;i&gt;Fzd6&lt;/i&gt; as potential target genes of &lt;i&gt;MiR-199b-5p&lt;/i&gt;. Thus, these results indicated that &lt;i&gt;MiR-199b-5p&lt;/i&gt;/&lt;i&gt;Gcnt2&lt;/i&gt; and &lt;i&gt;Fzd6&lt;/i&gt; axis might be a novel therapeutic target for the treatment of OA.</description>
      <author>wuqiaofeng@cdutcm.edu.cn (Fan-Rang Liang)</author>
      <author>wuqiaofeng@cdutcm.edu.cn (Jun-Meng Wang)</author>
      <author>wuqiaofeng@cdutcm.edu.cn (Mu-qiu Tian)</author>
      <author>wuqiaofeng@cdutcm.edu.cn (Qi Zhang)</author>
      <author>wuqiaofeng@cdutcm.edu.cn (Qiao-Feng Wu)</author>
      <author>wuqiaofeng@cdutcm.edu.cn (Shu-Guang Yu)</author>
      <author>wuqiaofeng@cdutcm.edu.cn (Shuan-hu Zhou)</author>
      <author>wuqiaofeng@cdutcm.edu.cn (Si-Hui Li)</author>
      <author>wuqiaofeng@cdutcm.edu.cn (Tong Feng)</author>
      <author>wuqiaofeng@cdutcm.edu.cn (Xin Wang)</author>
      <author>wuqiaofeng@cdutcm.edu.cn (Yan-ling Ping)</author>
      <guid isPermaLink="false">https://dx.doi.org/10.7554/eLife.92645</guid>
      <category>Immunology and Inflammation</category>
      <pubDate>Tue, 21 May 2024 00:00:00 +0000</pubDate>
      <dc:date>2024-05-21T00:00:00Z</dc:date>
      <webfeeds:featuredImage url="https://elife-cdn.s3.amazonaws.com/observer/elife-logo-408x230.svg" height="230" width="408" type="image/svg"/>
    </item>
[snip the rest]
```

The SQLite DB is 4.6GB. It would be good to rerun this once we have PostgreSQL setup by default, but at least we know it is running fine under this config containerised.

File: docker-compose.yaml [hunk 4]
```diff
@@ -0,0 +1,42 @@
...
+      retries: 3
+    build:
+      context: .
+    ports:
+    - 8000:8000
+    volumes:
+    - ./schema/metrics:/app/schema/metrics
+    - ./schema/sql:/app/schema/sql
+    - ./src:/app/src
+    - ./app.cfg:/app/app.cfg
```
[the remark this answers is not in the document]
Comment (ali-tm-amin) [reply docker-compose.yaml]: I think it looks better now, any other feedback, I will be more than happy to make a change

File: docker-compose.yaml [hunk 5]
```diff
@@ -0,0 +1,42 @@
...
+      test: pg_isready -d db_prod
+      interval: 10s
+      timeout: 3s
+      retries: 3
+    volumes:
+      - db_data:/var/lib/postgresql/data
+    ports:
+      - 5432:5432
+
+  localstack:
```
Comment (scottaubrey) [inline docker-compose.yaml:17]: I can't see this hostname referenced anywhere, so I'm not sure this is being used yet. However, we don't need it for now as no one is currently developing it, so I'm fine to figure that out later.

File: docs/docker-dev-setup.md [hunk 6]
```diff
@@ -0,0 +1,62 @@
...
+#### Stopping the services:
+```
+docker-compose down
+```
+Notes
+Ensure that the install.sh script has execute permissions before running the services:
+
+
+```
+chmod +x install.sh
```
Comment (scottaubrey) [inline docs/docker-dev-setup.md:35]: This is run as part of `manage.sh`, so doesn't need to be executable

**Review by scottaubrey — APPROVED** [review]
Some small nitpicks, but I'm OK with this going in as a starting point.

**Comment by scottaubrey** [comment]
And can confirm it's working and data is in the local postgres:

```shell
> docker compose exec app ./manage.sh load_from_api
[snip lots of output]

> curl -s http://localhost:8000/report/latest-articles | head -n 40
<?xml version='1.0' encoding='UTF-8'?>
<rss xmlns:dc="http://purl.org/dc/elements/1.1/" xmlns:webfeeds="http://webfeeds.org/rss/1.0" xmlns:atom="http://www.w3.org/2005/Atom" xmlns:content="http://purl.org/rss/1.0/modules/content/" version="2.0">
  <channel>
    <title>eLife: latest articles</title>
    <link>https://elifesciences.org</link>
    <description>All of the latest articles published at eLife, including in-progress POA (publish-on-accept) articles.</description>
    <atom:link href="https://observer.elifesciences.org/report/latest-articles" rel="self"/>
    <docs>http://www.rssboard.org/rss-specification</docs>
    <generator>observer (using python-feedgen)</generator>
    <language>en</language>
    <lastBuildDate>Sun, 26 May 2024 07:28:26 +0000</lastBuildDate>
    <webfeeds:analytics id="G-xxxxxxxxxx" engine="GoogleAnalytics"/>
    <item>
      <title>Secreted dengue virus NS1 from infection is predominantly dimeric and in complex with high-density lipoprotein</title>
      <link>https://elifesciences.org/articles/90762</link>
      <description>Severe dengue infections are characterized by endothelial dysfunction shown to be associated with the secreted nonstructural protein 1 (sNS1), making it an attractive vaccine antigen and biotherapeutic target. To uncover the biologically relevant structure of sNS1, we obtained infection-derived sNS1 (isNS1) from dengue virus (DENV)-infected Vero cells through immunoaffinity purification instead of recombinant sNS1 (rsNS1) overexpressed in insect or mammalian cell lines. We found that isNS1 appeared as an approximately 250 kDa complex of NS1 and ApoA1 and further determined the cryoEM structures of isNS1 and its complex with a monoclonal antibody/Fab. Indeed, we found that the major species of isNS1 is a complex of the NS1 dimer partially embedded in a high-density lipoprotein (HDL) particle. Crosslinking mass spectrometry studies confirmed that the isNS1 interacts with the major HDL component ApoA1 through interactions that map to the NS1 wing and hydrophobic domains. Furthermore, our studies demonstrated that the sNS1 in sera from DENV-infected mice and a human patient form a similar complex as isNS1. Our results report the molecular architecture of a biological form of sNS1, which may have implications for the molecular pathogenesis of dengue.</description>
      <author>luodahai@ntu.edu.sg (AN Qi Ngoh)</author>
      <author>luodahai@ntu.edu.sg (Bing Liang Alvin Chew)</author>
      <author>luodahai@ntu.edu.sg (Christiane Ruedl)</author>
      <author>luodahai@ntu.edu.sg (Dahai Luo)</author>
      <author>luodahai@ntu.edu.sg (Eng Eong Ooi)</author>
      <author>luodahai@ntu.edu.sg (Jenny Low)</author>
      <author>luodahai@ntu.edu.sg (Kitti Wing Ki Chan)</author>
      <author>luodahai@ntu.edu.sg (Mei Jie Grace Weng)</author>
      <author>luodahai@ntu.edu.sg (Milly M Choy)</author>
      <author>luodahai@ntu.edu.sg (Nikhil K Tulsian)</author>
      <author>luodahai@ntu.edu.sg (Radoslaw M Sobota)</author>
      <author>luodahai@ntu.edu.sg (Satoru Watanabe)</author>
      <author>luodahai@ntu.edu.sg (Shiao See Lim)</author>
      <author>luodahai@ntu.edu.sg (Subhash G Vasudevan)</author>
      <author>luodahai@ntu.edu.sg (Wint Wint Phoo)</author>
      <author>luodahai@ntu.edu.sg (Zheng Ser)</author>
      <guid isPermaLink="false">https://dx.doi.org/10.7554/eLife.90762</guid>
      <category>Microbiology and Infectious Disease</category>
      <category>Structural Biology and Molecular Biophysics</category>
      <pubDate>Fri, 24 May 2024 00:00:00 +0000</pubDate>
      <dc:date>2024-05-24T00:00:00Z</dc:date>
      <webfeeds:featuredImage url="https://elife-cdn.s3.amazonaws.com/observer/elife-logo-408x230.svg" height="230" width="408" type="image/svg"/>
    </item>
    <item>

> docker compose exec postgres psql -U postgres
psql (16.3 (Debian 16.3-1.pgdg120+1))
Type "help" for help.

postgres=# \dt
                     List of relations
 Schema |              Name              | Type  |  Owner   
--------+--------------------------------+-------+----------
 public | articles                       | table | postgres
 public | articles_authors               | table | postgres
 public | articles_subjects              | table | postgres
 public | auth_group                     | table | postgres
 public | auth_group_permissions         | table | postgres
 public | auth_permission                | table | postgres
 public | auth_user                      | table | postgres
 public | auth_user_groups               | table | postgres
 public | auth_user_user_permissions     | table | postgres
 public | django_admin_log               | table | postgres
 public | django_content_type            | table | postgres
 public | django_migrations              | table | postgres
 public | django_session                 | table | postgres
 public | observer_author                | table | postgres
 public | observer_content               | table | postgres
 public | observer_content_categories    | table | postgres
 public | observer_contentcategory       | table | postgres
 public | observer_presspackage          | table | postgres
 public | observer_presspackage_subjects | table | postgres
 public | observer_profile               | table | postgres
 public | observer_rawjson               | table | postgres
 public | observer_subject               | table | postgres
(22 rows)

postgres=# SELECT msid, doi FROM articles WHERE msid=85111;
 msid  |         doi         
-------+---------------------
 85111 | 10.7554/eLife.85111
(1 row)
```

**Pull request merged**